src/renderer/utils/FileMonitor.tsx

```diff
@@ -0,0 +1,264 @@
+import chokidar from 'chokidar';
+import { Subject } from 'rxjs';
+
+import { Event, Monitor } from './Monitor';
+
+// type Event = 'unlink' | 'unlinkDir' | 'add' | 'addDir' | 'change' | 'error';
+
+class FileMonitor implements Monitor {
+  //The main functionality of the file monitor is to report selected type of events using the Observable called subject
+  private folderPath: string;
+  //ignoredPath: string;
+  private events: Array<Event>;
+  private eventsWatching: Array<Event>;
+  private subject: Subject<[Event, string]>;
+  private addWatcher: chokidar.FSWatcher;
+  private addDirWatcher: chokidar.FSWatcher;
+  private unlinkWatcher: chokidar.FSWatcher;
+  private unlinkDirWatcher: chokidar.FSWatcher;
+  private changeWatcher: chokidar.FSWatcher;
+  private errorWatcher: chokidar.FSWatcher;
+
+  //construct with the folder path that you want to monitor and ignoredPath that the watcher will ignore
+  constructor(folderPath: string) {
+    this.subject = new Subject();
+    this.folderPath = folderPath;
+    //this.ignoredPath = ignoredPath;
+    this.events = [];
+    this.eventsWatching = [];
+    this.addWatcher = chokidar.watch(this.folderPath, {
+      ignored: [/(^|[/\\])\../],
+      followSymlinks: false,
+      persistent: true,
+    });
+    this.addDirWatcher = chokidar.watch(this.folderPath, {
+      ignored: [/(^|[/\\])\../],
+      followSymlinks: false,
+      persistent: true,
+    });
+    this.unlinkWatcher = chokidar.watch(this.folderPath, {
+      ignored: [/(^|[/\\])\../],
+      followSymlinks: false,
+      persistent: true,
+    });
+    this.unlinkDirWatcher = chokidar.watch(this.folderPath, {
+      ignored: [/(^|[/\\])\../],
+      followSymlinks: false,
+      persistent: true,
+    });
+    this.changeWatcher = chokidar.watch(this.folderPath, {
+      ignored: [/(^|[/\\])\../],
+      followSymlinks: false,
+      persistent: true,
+    });
+    this.errorWatcher = chokidar.watch(this.folderPath, {
+      ignored: [/(^|[/\\])\../],
+      followSymlinks: false,
+      persistent: true,
+    });
+  }
+
+  //Input: array of events
+  //note that the events can be added to the watcher but CANNOT be removed from watcher
+  public watch(events: Array<Event>): void {
+    this.events.push(...events);
+    //unlink: files being deleted
+    if (this.events.includes('unlink') && !this.eventsWatching.includes('unlink')) {
```

Comment:
A lot of repetitive code here. Can you simplify it using loops, maps, and helper functions?